### PR TITLE
[Core][Tests] Cleanup HTTPServer start and stop

### PR DESCRIPTION
**Pull request by Liquid369**
Lately, we have been spending a lot of time messing with our test cases. We are hitting random errors at various points and it is causing us to waste time constantly fixing and re-running tests in order to be 'ready for review'. After discussion with the other developers, I went and looked into the code and found some tiny improvements that can assist us in not dealing with these cases anymore.

Threads are still considered active, even if the code is finished executing before being joined, and we do not want to try to join them if the code is not finished executing. So we are now checking if the threads can be closed and we are doing a little bit better cleanup of 0 > nullptr. Lastly, we want to make sure any queue is empty before closing.

To test: 
The best route is to fork this branch and base it over your master or any branch you can run GitHub actions on, and run the tests several times, as well as running the test-runner.py on your local PC. 

I have run over 50+ iterations of the tests over the weekend and have not encountered an issue in shutdown since.
We still have some issues in `tier_two` tests that may appear, but evaluate the logs for whether or not you are incurring an issue from the `stop_node()` function.

### CODE REVIEW

**Review by panleone — COMMENTED** [review]
Just one comment but overall I like the PR

File: src/httpserver.cpp [hunk 1]
```diff
@@ -110,7 +110,7 @@ class WorkQueue
                 std::unique_lock<std::mutex> lock(cs);
                 while (running && queue.empty())
                     cond.wait(lock);
-                if (!running)
+                if (!running && queue.empty())
```
[the remark this answers is not in the document]
Comment (Liquid369) [reply src/httpserver.cpp:113]: Okay, I agree we should also avoid queueing if we are not yet running!

**Review by Fuzzbawls — APPROVED** [review]
ACK 9d423606dc12050b401cd4528a3239db37ea7e16

Looking good on this one.

**Review by panleone — APPROVED** [review]
tACK 9d423606dc12050b401cd4528a3239db37ea7e16

**Pull request merged**